Sources/StatsdClient/StatsdClient.swift

```diff
@@ -349,11 +361,61 @@ private final class Client {
     }
 }
 
+// MARK: - Label Sanitizer
+
+/// Used to sanitize labels (and dimensions) into a format compatible with statsd's wire format.
+///
+/// By default `StatsdClient` uses the `StatsdLabelSanitizer`.
+public protocol LabelSanitizer {
```

Comment:
Do we ever forsee needing a second one of these?

Reply:
They come in handy when "library x" uses keys which read badly when escaped but you have a good idea how you'd like to represent them.

Specifically in this example there's a project which uses Vapor and Vapor today uses labels such as: 

`/hello/:who` so the `:` breaks things, and while our "safe default substitution" to `/hello/_who` reads "ok" users may prefer to have those show up as `/hello/who` in their metrics backends or something similar. Thus leaving it open.

This also follows prior art as the Prometheus metrics backend is doing; We also debated there if it's necessary, but I guess consistency is a nice thing to have for those.

Reply:
Do you think we should be worried about the performance overhead? In my use cases metrics objects are created and reused so it's less worrying if sanitizing the label has to call through a protocol... but perhaps there's users which "always" would `Counter("silly:label")`...

Reply:
I'm not so much worried about the performance overhead as I am worried about overgeneralising. Defining protocols where they aren't needed is awkward.

That said, given that there's prior art, I'm not going to stand in the way of this. That said, I do think we have a problem with the protocol names. This is the second module that defines `LabelSanitizer` as a public name, meaning that anyone who imports both will get ambiguous type errors if they attempt to conform. That's annoying. If we think this is a pattern all the metrics libraries should follow, we should arrange to ensure that part of the pattern is that these protocols all have different names!

Probably this should be called `StatsdLabelSanitizer`.

Reply:
I see, yeah that's a good point about the names. While I'm not really expecting systems to use multiple metrics backends, there always could be some legitimate reason for doing so.

I like the naming proposal, so it'll be `StatsdLabelSanitizer` the protocol and `DefaultStatsdLabelSanitizer` the impl.

Reply:
Alternatively we could call them `StatsdMetricNameSanitizer`, technically that's what the field this sanitizes in statsd wording is. 

The only difference I can see is a slight semantic one:
- label is label from swift-metrics
- we _also_ have to sanitize the entire metric name
  - the entire metric name equals label + dimensions

So it would be a bit more correct to call it `StatsdMetricNameSanitizer`... Maybe Tom will have an opinion on this 🤔

Reply:
Sounds good, simplified here: https://github.com/apple/swift-statsd-client/pull/22/commits/2910f13b63d089e6ba88df13e500d02bf82edbeb

It's just a typealias now.

I just added the (defaulted) parameter to the existing init since we're in "alpha" versions.